DOCSP-30513: Insert Many Usage Example (#74)
mongodb/docs-rust: source/usage-examples/insertMany.txt

.. _rust-insert-many-usage:

=========================
Insert Multiple Documents
=========================

You can insert multiple documents into a collection by calling the
`insert_many() <{+api+}/struct.Collection.html#method.insert_many>`__ method
on a ``Collection`` instance.

Pass a vector containing one or more documents to the ``insert_many()`` method
to insert them into your collection. If you have only one document to insert,
consider using the `insert_one() <{+api+}/struct.Collection.html#method.insert_one>`__

> method instead.

Reviewer: S: In this section, can you add some information about collection parameterization [as I have done here](https://github.com/mongodb/docs-rust/pull/75/files#diff-d30c201a190c69dc54f954f9d80b9bd3c432a06908dc5d9a075752e91a9d6a49R11-R17) in the insert_one usage ex?

> method instead.

Reviewer: S: move the information about `insert_one()` into a tip

> The ``insert_many()`` method returns a `Result<InsertManyResult> <{+api+}/results/struct.InsertManyResult.html>`__
> type from which you can access the ``_id`` values of the inserted documents.

Reviewer: S: The `?` operator that you include at the end of the insert statement "unwraps" the Result into the actual type, so no need to include it.
```suggestion
The ``insert_many()`` method returns an `InsertManyResult <{+api+}/results/struct.InsertManyResult.html>`__
type from which you can access the ``_id`` values of the newly inserted documents.
```

Author: Good catch, fixed!

.. tip::

   To learn more about inserting documents into a collection, see the

>    :ref:`rust-insert-guide` guide.

Reviewer: Move this out of the tip, it is not really extra information

Example
-------

This example inserts documents into the ``restaurants`` collection of the
``sample_restaurants`` database. The example uses a ``Restaurant`` struct containing

> ``name`` and ``cuisine`` fields to model the documents that are inserted into the
> collection.

Reviewer: ```suggestion
``name`` and ``cuisine`` fields to model the documents being inserted into the collection.
```

The following code passes a vector of documents as a parameter to the
``insert_many()`` method. The vector contains documents in which the value of the

> ``cuisine`` field is ``"Nepalese"``.

Reviewer: S: remove the second sentence

> Select the **Asynchronous** or **Synchronous** tab to see corresponding code for each runtime:

Reviewer: Q: should we add this to all of the usage examples?

Author: I think it's good to have

.. tabs::

   .. tab:: Asynchronous
      :tabid: find-async

      .. io-code-block::
         :copyable: true

        .. input:: /includes/usage-examples/code-snippets/insert-many-async.rs
            :language: rust
            :dedent:

        .. output::
            :language: console
            :visible: false

>             // Your ObjectID values might differ
>             Inserted documents with _ids:
>             ObjectId("654516036a974e45a716c458")
>             ObjectId("654516036a974e45a716c45a")

Reviewer: S: instead of having the comment here, just truncate the ObjectIds
```suggestion
            // Your ObjectID values might differ
            Inserted documents with _ids:
            ObjectId("...")
            ObjectId("...")
```

Reviewer: same for the other tab

.. _rust-insert-many-usage:

=========================
Insert Multiple Documents
=========================

You can insert multiple documents into a collection by calling the
`insert_many() <{+api+}/struct.Collection.html#method.insert_many>`__ method
on a ``Collection`` instance.

Pass a vector containing one or more documents to the ``insert_many()`` method
to insert them into your collection. These documents must be instances of the type
that you parameterized your ``Collection`` instance with. For example, if you
parameterized your collection with the ``MyStruct`` struct, pass a vector of ``MyStruct``
instances as a parameter to the ``insert_many()`` method. 

> .. tip:: 
> 
>    If you have only one document to insert, consider using the `insert_one() 
>    <{+api+}/struct.Collection.html#method.insert_one>`__ method instead.

Reviewer: S: wording and also, I just merged insert one usage ex so you can merge master into this branch and then update this admonition to link to that usage example ie "To see an example that uses this method, see the ... usage example"

```suggestion
.. tip:: 

   To insert a single document, consider using the `insert_one() 
   <{+api+}/struct.Collection.html#method.insert_one>`__ method instead.
```

> The ``insert_many()`` method returns an `InsertManyResult <{+api+}/results/struct.InsertManyResult.html>`__
> type from which you can access the ``_id`` values of the inserted documents.

Reviewer: ```suggestion
The ``insert_many()`` method returns an `InsertManyResult <{+api+}/results/struct.InsertManyResult.html>`__
type that references the ``_id`` values of the inserted documents.
```
